packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -1058,3 +1065,273 @@ class CallbackShortcuts extends StatelessWidget {
+    };
+  }
+  final Map<ShortcutRegistryToken, Map<ShortcutActivator, Intent>> _tokenShortcuts =
+    <ShortcutRegistryToken, Map<ShortcutActivator, Intent>>{};
+
+  /// Adds all the given shortcut bindings to this [ShortcutRegistry], and
+  /// returns a token for managing those bindings.
+  ///
+  /// The token should have [ShortcutRegistryToken.dispose] called on it when
+  /// these shortcuts are no longer needed. This will remove them from the
+  /// registry, and invalidate the token.
+  ///
+  /// This method will assert in debug mode if another token exists (i.e. hasn't been
+  /// disposed of) has already added a given shortcut.
+  ///
+  /// If two equivalent, but different, [ShortcutActivator]s are added, all of
+  /// them will be executed when triggered. For example, if both
+  /// `SingleActivator(LogicalKeyboardKey.keyA)` and `CharacterActivator('a')`
+  /// are added, then both will be executed when an "a" key is pressed.
+  ShortcutRegistryToken addAll(Map<ShortcutActivator, Intent> value) {
+    assert(debugAssertNotDisposed());
+    final ShortcutRegistryToken token = ShortcutRegistryToken._(this);
+    _tokenShortcuts[token] = value;
+    _debugCheckForDuplicates();
+    _indexedShortcutsCache = null;
+    notifyListeners();
+    return token;
+  }
+
+  /// Replaces the given shortcut bindings in this [ShortcutRegistry].
+  ///
+  /// This method will assert in debug mode if another [ShortcutRegistryToken]
+  /// exists (i.e. hasn't been disposed of) that has already added a given
+  /// shortcut.
+  ///
+  /// It will also assert if the given token wasn't created by this registry, or
+  /// has already been disposed.
+  ///
+  /// If two equivalent, but different, [ShortcutActivator]s are added, all of
+  /// them will be executed when triggered. For example, if both
+  /// `SingleActivator(LogicalKeyboardKey.keyA)` and `CharacterActivator('a')`
+  /// are added, then both will be executed when an "a" key is pressed.
+  void replaceAll(ShortcutRegistryToken token, Map<ShortcutActivator, Intent> value) {
+    assert(debugAssertNotDisposed());
+    assert(_tokenShortcuts.containsKey(token),
+      'Token $token is not present in this ShortcutRegistry.\n'
+      'The token ${token.registry == this
+          ? 'was created by this registry, but has probably already been disposed of'
+          : 'was not created by this registry, it was created by ${token.registry}'}.');
+    _tokenShortcuts[token] = value;
+    _indexedShortcutsCache = null;
+    _debugCheckForDuplicates();
+    notifyListeners();
+  }
+
+
+  /// Returns the [ShortcutRegistry] that belongs to the [ShortcutRegistrar]
+  /// which most tightly encloses the given [BuildContext].
+  ///
+  /// If no [ShortcutRegistrar] widget encloses the context given, `of` will
+  /// throw an exception in debug mode.
+  ///
+  /// The dependencies of [ShortcutRegistrar] will have their
+  /// [State.didChangeDependencies] called whenever the shortcuts have changed.
```

Comment:
Is this still true? Dependencies will be informed when the identity of the ShortcutRegistry changes, but if they want to know that shortcuts have changed, they need to add a listener to it, no?

Reply:
Right. Updated.